py_zerox/zerox/models/modellitellm.py

```diff
@@ -1,50 +1,82 @@
+        **kwargs,
     ):
-        """Initializes the OpenAI model."""
-        if api_key is None:
-            api_key = self.get_api_key()
-
-        super().__init__(api_key)
-
-    @staticmethod
-    def get_api_key(
-        api_key: Optional[str] = None,
-    ) -> str:
-        """Gets the OpenAI API key from the environment variables incase it is not provided."""
-        if api_key is None:
-            api_key = os.getenv("OPENAI_API_KEY")
-        if api_key is None:
-            raise MissingOpenAIAPIKeyException()
-        return api_key
+        """
+        Initializes the Litellm model interface.
+        :param model: The model to use for generating completions, defaults to "gpt-4o-mini". Refer: https://docs.litellm.ai/docs/providers
+        :type model: str, optional
+        
+        :param kwargs: Additional keyword arguments to pass to self.completion -> litellm.completion. Refer: https://docs.litellm.ai/docs/providers and https://docs.litellm.ai/docs/completion/input
+        """
+        super().__init__(model=model, **kwargs)
+
+        ## calling custom methods to validate the environment and model
+        self.validate_environment()
+        self.validate_model()
+        self.validate_access()
+
+    @property
+    def system_prompt(self) -> str:
+        '''Returns the system prompt for the model.'''
+        return self._system_prompt
+    
+    @system_prompt.setter
+    def system_prompt(self, prompt: str) -> None:
+        '''
+        Sets/overrides the system prompt for the model.
+        Will raise a friendly warning to notify the user. 
+        '''
+        warnings.warn(f"{Messages.CUSTOM_SYSTEM_PROMPT_WARNING}. Default prompt for zerox is:\n {DEFAULT_SYSTEM_PROMPT}")
+        self._system_prompt = prompt
+
+    ## custom method on top of BaseModel
+    def validate_environment(self) -> None:
+        """Validates the environment variables required for the model."""
+        env_config = litellm.validate_environment(model=self.model)
+
+        if not env_config["keys_in_environment"]:
+            raise MissingEnvironmentVariables(extra_info=env_config)
+        
+    def validate_model(self) -> None:
+        '''Validates the model to ensure it is a vision model.'''
+        if not litellm.supports_vision(model=self.model):
+            raise NotAVisionModel(extra_info={"model": self.model})
+        
+    def validate_access(self) -> None:
+        """Validates access to the model -> if environment variables are set correctly with correct values."""
+        if not litellm.check_valid_key(model=self.model,api_key=None):
+            raise ModelAccessError(extra_info={"model": self.model})
+        
 
     async def completion(
         self,
         image_path: str,
         maintain_format: bool,
         prior_page: str,
-        model: str = "gpt-4o-mini",
+        **kwargs,
     ) -> CompletionResponse:
         """OpenAI completion for image to markdown conversion.
```

Comment:
OpenAI completion -> LLM completion

Reply:
@xdotli, Just realized, in this kwargs wouldn't be passed explicilty, but instead using object's attribute **self.kwargs. Will fix it. Thanks.

Reply:
fixed in [commit cad456b](https://github.com/getomni-ai/zerox/pull/21/commits/cad456bf509c6691064fef36c136d9cbf9ae157e)